### PR TITLE
[Merged by Bors] - feat(FieldTheory): define induced action of subfields and intermediate fields

### PR DESCRIPTION
**Pull request by jjdishere**
Add instances for induced actions of subfields and intermediate fields.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Field/Subfield.lean [hunk 1]
```diff
@@ -873,3 +873,69 @@ theorem comap_map (f : K →+* L) (s : Subfield K) : (s.map f).comap f = s :=
...
+`Subsemiring.MulAction`.
+-/
+section Actions
+
+namespace Subfield
+
+variable {X Y}
+
+/-- The action by a subfield is the action by the underlying field. -/
+instance [SMul K X] (F : Subfield K): SMul F X :=
```
Comment (tb65536) [inline Mathlib/Algebra/Field/Subfield.lean:889]:
```suggestion
instance [SMul K X] (F : Subfield K) : SMul F X :=
```

File: Mathlib/FieldTheory/IntermediateField/Basic.lean [hunk 2]
```diff
@@ -302,6 +302,64 @@ theorem coe_prod {ι : Type*} [Fintype ι] (f : ι → S) : (↑(∏ i, f i) : L
...
     · rw [Finset.prod_insert hi, MulMemClass.coe_mul, H, Finset.prod_insert hi]
 
+/-!
+`IntermediateField`s inherit structure from their `Subfield` coercions.
+-/
+
+variable {X Y}
+
+/-- The action by an intermediate field is the action by the underlying field. -/
+instance [SMul L X] (F : IntermediateField K L): SMul F X :=
```
Comment (tb65536) [inline Mathlib/FieldTheory/IntermediateField/Basic.lean:312]:
```suggestion
instance [SMul L X] (F : IntermediateField K L) : SMul F X :=
```

File: Mathlib/Algebra/Field/Subfield.lean (hunk 1 above)
Comment (jjdishere) [reply Mathlib/Algebra/Field/Subfield.lean:889]: Thank you!

**Review by YaelDillies — APPROVED** [review]
Straightforward, thanks!

maintainer merge

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**